### PR TITLE
doc: Update intro with ble mesh and kernel features

### PR DESCRIPTION
**Pull request by nashif**
Expand introduction about Zephyr with additional details.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Comment by nashif** [comment]
@dbkinder ping :)

File: doc/introduction/introducing_zephyr.rst [hunk 1]
```diff
@@ -29,8 +30,42 @@ as described in :ref:`Zephyr_Licensing`.
...
+     counting semaphores, and mutex semaphores.
+
+   * *Inter-thread Data Passing Services* for basic message queues, enhanced
+     message queues, and byte streams.
+
+   * *Power Management Services* such as tickless idle and an advanced idling
+     infrastructure.
+
+**Multiple Scheduling Algorithms**
+
```
Comment (dbkinder) [inline doc/introduction/introducing_zephyr.rst]: This section is a series of high-level topics using a reST dictionary list.  To have this topic look like the rest of the topics, it needs to start with an indented sentence to contain the nested bullet points. So, change this to:

    **Multiple Scheduling Algorithms**

       Zephyr provides a comprehensive set of thread scheduling choices:

       * Cooperative and Preemptive Scheduling

File: doc/introduction/introducing_zephyr.rst [hunk 2]
```diff
@@ -90,6 +125,12 @@ small-footprint OSes:
    * Verified with multiple popular controllers
    * Highly configurable
 
+   Mesh Support:
+
+   * Relay, Friend Node, Low-Power Node (LPN) and GATT Proxy features
+   * Both Provisioning bearers supported (PB-ADV & PB-GATT)
+   * Highly configurable, fits as small as 16k RAM devices
```
Comment (dbkinder) [inline doc/introduction/introducing_zephyr.rst]: How about:

    * Highly configurable, fitting in small 16Kb RAM devices

File: doc/introduction/introducing_zephyr.rst [hunk 3]
```diff
@@ -4,11 +4,12 @@ Introducing Zephyr
 ##################
 
 The Zephyr OS is based on a small-footprint kernel designed for use on
-resource-constrained systems: from simple embedded environmental sensors and LED
-wearables to sophisticated smart watches and IoT wireless gateways.
+resource-constrained and embedded systems: from simple embedded environmental
+sensors and LED wearables to sophisticated embedded controllers, smart
+watches and IoT wireless applications.
```
Comment (dbkinder) [inline doc/introduction/introducing_zephyr.rst]: Since I had other suggestions, I'll mention a missing comma here:

    watches, and ...

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/introduction/introducing_zephyr.rst [hunk 4]
```diff
@@ -29,8 +30,44 @@ as described in :ref:`Zephyr_Licensing`.
...
+     counting semaphores, and mutex semaphores.
+
+   * *Inter-thread Data Passing Services* for basic message queues, enhanced
+     message queues, and byte streams.
+
+   * *Power Management Services* such as tickless idle and an advanced idling
+     infrastructure.
+
+**Multiple Scheduling Algorithms**
+
```
Comment (dbkinder) [inline doc/introduction/introducing_zephyr.rst]: My bad... delete this blank line (57)  to have the dictionary list display correctly.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by dbkinder — APPROVED** [review]
Thanks

**Pull request merged**